test/compilable/test19203.d

```diff
@@ -0,0 +1,27 @@
+//https://issues.dlang.org/show_bug.cgi?id=19203
+struct BoolWithErr {
+    bool b;
+    string error;
+    alias b this;
+}
+
+struct Foo {
+    int popBack() { return 0; }
+}
+
+struct Bar {}
+
+template hasPopBack(T) {
+    static if (!is(typeof(T.init.popBack)))
+        enum hasPopBack = BoolWithErr(false, T.stringof~" does not have popBack");
+    else
+        enum hasPopBack = BoolWithErr(true,"");
+}
+
+void test()
+{
+    static assert( hasPopBack!Foo);
+    static assert(!hasPopBack!Bar);
+    static assert( hasPopBack!Foo && !hasPopBack!Bar);
```

Comment:
Because before it would complain with `foo && bar` if both were `alias this` to a bool then the result of the expression is a bool, but would fail because `foo` was not a bool.

Reply:
See the bug report for more detail.